FlashLFQ/FlashLfqEngine.cs

```diff
@@ -44,19 +51,30 @@ public class FlashLfqEngine
         private Dictionary<SpectraFileInfo, Ms1ScanInfo[]> _ms1Scans;
         private PeakIndexingEngine _peakIndexingEngine;
 
-        public FlashLfqEngine(List<Identification> allIdentifications, bool normalize = false,
-            bool advancedProteinQuant = false, bool matchBetweenRuns = false,
-            double ppmTolerance = 10.0, double isotopeTolerancePpm = 5.0, double matchBetweenRunsPpmTolerance = 5.0,
-            bool integrate = false, int numIsotopesRequired = 2,
-            bool idSpecificChargeState = false, bool requireMonoisotopicMass = true, bool silent = false,
-            string optionalPeriodicTablePath = null, double maxMbrWindow = 2.5,
-            int maxThreads = -1)
+        public FlashLfqEngine(
+            List<Identification> allIdentifications,
+            bool normalize = false,
+            bool bayesianProteinQuant = false,
+            bool matchBetweenRuns = false,
+            double ppmTolerance = 10.0,
+            double isotopeTolerancePpm = 5.0,
+            double matchBetweenRunsPpmTolerance = 5.0,
+            bool integrate = false,
+            int numIsotopesRequired = 2,
+            bool idSpecificChargeState = false,
+            bool requireMonoisotopicMass = true,
+            bool silent = false,
+            double maxMbrWindow = 2.5,
+            int maxThreads = -1,
+
+            // settings for the Bayesian protein quantification engine
+            string proteinQuantBaseCondition = null,
+            double? proteinQuantFoldChangeCutoff = null,
+            int mcmcSteps = 3000,
+            int mcmcBurninSteps = 1000,
+            bool useSharedPeptidesForProteinQuant = false,
+            int? randomSeed = null)
```

Comment:
Is there a reason to make this nullable? Is it here for people that prefer their data to not be reproducible?

Reply:
a random seed is automatically generated if one isn't provided. the seed should be `public` instead of `private`, though, for it to be reported along with the output, so that you can reproduce someone else's work. but it isn't necessary to generate your own random seed